### PR TITLE
Starting playground for dedalus+pySDC

### PR DESCRIPTION
**Pull request by tlunet**
This PR include the latest developments for combining pySDC and dedalus, in particular :

- a `DedalusSweeperIMEX` and `DedalusProblem` that can be used within a pySDC controller to solve any IVP Dedalus problem.
- a `SpectralDeferredCorrectionIMEX` time-integrator class that can be used as any other dedalus time-stepping class.

Since this is work in progress, there is no testing yet of those classes, even if results can be visually compared with a standard time-integration method from dedalus on a non-linear Burger problem (generated figures are visually similar). Hence they are located for now in a playground.

This PR is intended to trigger a new minor release of pySDC to be added to a project report.

### CODE REVIEW

File: pySDC/playgrounds/dedalus/demo_advection.png
Comment (brownbaerchen) [inline pySDC/playgrounds/dedalus/demo_advection.png:1]: Did you intentionally add this image to the commit?

**Comment by pancetta** [comment]
Niceee.. want me to merge and create a new release?

File: pySDC/playgrounds/dedalus/demo_advection.png
Comment (pancetta) [reply pySDC/playgrounds/dedalus/demo_advection.png:1]: Looks like this is part of the readme?

Comment (tlunet) [reply pySDC/playgrounds/dedalus/demo_advection.png:1]: Yes indeed : allows to see the image directly with the github link, see https://github.com/tlunet/pySDC/tree/dedalus/pySDC/playgrounds/dedalus

**Comment by pancetta** [comment]
Go ahead, then!

**Pull request merged**